security/s2a/internal/crypter/aesgcm_test.go

```diff
@@ -0,0 +1,474 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package crypter
+
+import (
+	"bytes"
+	"fmt"
+	"google.golang.org/grpc/security/s2a/internal/crypter/testutil"
+	"testing"
+)
+
+// getGCMCryptoPair outputs a sender/receiver pair on AES-GCM.
+func getGCMCryptoPair(key []byte, t *testing.T) (S2AAeadCrypter, S2AAeadCrypter) {
+	sender, err := NewAESGCM(key)
+	if err != nil {
+		t.Fatalf("NewAESGCM(ClientSide, key) = %v", err)
+	}
+	receiver, err := NewAESGCM(key)
+	if err != nil {
+		t.Fatalf("NewAESGCM(ServerSide, key) = %v", err)
+	}
+	return sender, receiver
+}
+
+func isFailure(result string, err error, got, expected []byte) bool {
+	return (result == testutil.ValidResult && (err != nil || !bytes.Equal(got, expected))) ||
+		(result == testutil.InvalidResult && bytes.Equal(got, expected))
+}
+
+func wycheProofTestVectorFilter(testGroup testutil.TestGroup) bool {
+	return testGroup.IVSize != 96 ||
+		(testGroup.KeySize != 128 && testGroup.KeySize != 256) ||
+		testGroup.TagSize != 128
+}
+
+func testGCMEncryptionDecryption(sender S2AAeadCrypter, receiver S2AAeadCrypter, test *testutil.CryptoTestVector, t *testing.T) {
+	// Ciphertext is: encrypted text + tag.
```

Comment:
Please use `ciphertext`, with a small `c`, to make it clear that it's referring to the new variable we're defining in the next line.

Reply:
Done